Slim/Plugin/UPnP/MediaRenderer/AVTransport.pm

```diff
@@ -99,6 +99,45 @@ sub clientEvent {
 		return;
 	}
 
+	# Player jumped to new song, check if NextURI needs to be moved to CurrentURI
+	# for gapless playback support.
+	if ( $cmd eq 'newsong' ) {
+		my $playlist = Slim::Player::Playlist::playList($client);
+		if ( scalar @{$playlist} > 1 ){
+			if ( my $song = ($client->playingSong() || $client->streamingSong()) ) {
+
+				my $pd = $client->pluginData();
+				my $currentURI = $pd->{AVT}->{CurrentTrackURI};
+				my $currentURIMetadata = $pd->{AVT}->{CurrentTrackMetaData};
+				my $track = $song->currentTrack;
+
+				# Convert URI to protocol handler
+				$currentURI =~ s/^http/upnp/; 
+
+				if ( $track->url ne $currentURI ){
+
+					# player has moved on to next track, copy NextURI to CurrentURI
+					$pd->{avt_AVTransportURIMetaData_hash} = $pd->{avt_NextAVTransportURIMetaData_hash};
+					$currentURI = $pd->{AVT}->{NextAVTransportURI};
+					$currentURIMetadata = $pd->{AVT}->{NextAVTransportURIMetaData};
+
+					# Remove track that just finished playing. The protocol handler
+					# can only return metadata for currentURI and NextURI, so keep
+					# the playlist limited to those 2 tracks
+					$client->execute( [ 'playlist', 'delete', 0 ] );
+
+					# change state variables
+					$class->changeState( $client, {
+						CurrentTrackURI				=> $currentURI,
+						CurrentTrackMetaData		=> $currentURIMetadata,
+						AVTransportURI				=> $currentURI,
+						AVTransportURIMetaData		=> $currentURIMetadata,
+					} );
+				}
+			}
+		}
+	}
```

Comment:
This change unfortunately break "regular" playback for native LMS clients. See https://forums.lyrion.org/forum/user-forums/logitech-media-server/1764092: with this plugin enabled playback would fail. With the plugin disabled (or these lines removed) LMS is working as expected again.

Deleting played items from the playlist probably are the wrong thing to do. Why would you not be able to get metadata for any track? I think that's an unnecessary limitation in this change set, as we do fetch metadata for all tracks all the time.

Could you please look into this? I might revert this change for the time being until we've found a better way to achieve what you're trying to achieve. Thanks for your understanding!